### ISSUE TITLE
"Focus points" for images

### ISSUE THREAD

**Issue by ggttttggggtt** [Hyperchroma/hyperchroma#6]
**Describe the feature**
The ability to set one or multiple focus points per image by drawing a frame that the camera will attempt to stay inside of, or gravitate to. Alternatively, the ability to draw a frame that the camera will avoid.

**Examples**
![image](https://user-images.githubusercontent.com/87790164/158042098-5ae695c4-0f85-492b-9f86-aed56894c3ce.png)

Something like this.

**Additional context**
Sometimes the camera will focus on boring or less interesting parts of an image, especially if it's portrait and the window is landscape, vice versa.

**Comment by aeroheim** [comment]
Are you thinking of this as a quick override or as something done manually for many images? This could be useful for the former, editing the actual preset would probably be better for the latter. 

If you're using the default preset, it should be under the "Slow Pan" and "Fast Pan" effects. The configuration defines a horizontal and vertical range allowed for the camera movements.

**Comment by ggttttggggtt** [comment]
I was thinking of it as a manual override for each specific image. Editing the preset directly works if all images are the same aspect ratio, but if they aren't then the "acceptable" ranges will vary a lot. And editing it without a visual display of the frame is a bit cumbersome.

I imagined it as a context option in this menu, something like this maybe and when selected it opens the image in a dialog where you can draw a frame.
![image](https://user-images.githubusercontent.com/87790164/158744954-b775e2b2-3d60-49b1-8323-a006937ba024.png)

**Comment by aeroheim** [comment]
Yeah, working with arbitrary window and image sizes is a disadvantage with the current camera configurations. 

I have a planned feature to add a saliency detection algorithm to make the camera positioning more intelligent. Ideally it should be robust enough so that users won't need to worry about specifying manual overrides. 

Once I get around to I'll take a look at this again and see if this still makes sense or not to add.
